### PR TITLE
CI matrix change: ubuntu

### PR DESCRIPTION
**Pull request by aik-jahoda**
Changes in CI matrix:

| OS | What to do | Notes
| - | - | -
| Ubuntu 16.04 LTS | Remove PR coverage when used without a container.

part of #57947

### CODE REVIEW

File: eng/pipelines/coreclr/templates/helix-queues-setup.yml [hunk 1]
```diff
@@ -91,7 +91,6 @@ jobs:
       - ${{ if and(eq(variables['System.TeamProject'], 'public'), notIn(parameters.jobParameters.helixQueueGroup, 'pr', 'ci', 'libraries')) }}:
         - (Debian.10.Amd64.Open)Ubuntu.1804.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-10-helix-amd64-20210304164434-56c6673
         - (Debian.11.Amd64.Open)Ubuntu.1804.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-11-helix-amd64-20210304164428-5a7c380
-        - Ubuntu.1604.Amd64.Open
```
Comment (safern) [inline eng/pipelines/coreclr/templates/helix-queues-setup.yml:94]: There are more entries missing (on the next if statement).

File: eng/pipelines/libraries/helix-queues-setup.yml [hunk 2]
```diff
@@ -79,7 +79,6 @@ jobs:
             - RedHat.7.Amd64.Open
             - (Debian.10.Amd64.Open)Ubuntu.1804.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-10-helix-amd64-20210304164434-56c6673
             - (Debian.11.Amd64.Open)Ubuntu.1804.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-11-helix-amd64-20210304164428-5a7c380
-            - Ubuntu.1604.Amd64.Open
```
Comment (safern) [inline eng/pipelines/libraries/helix-queues-setup.yml:82]: There are some other instances missing (the if above).

File: eng/pipelines/coreclr/templates/helix-queues-setup.yml (hunk 1 above)
Comment (aik-jahoda) [reply eng/pipelines/coreclr/templates/helix-queues-setup.yml:94]: The next if statement is for 'internal' queue which is not running. The whole 'internal' should be removed (#57068). I will remove it for completeness.

Comment (safern) [reply eng/pipelines/coreclr/templates/helix-queues-setup.yml:94]: I thought at some point we wanted to enable internal test runs for servicing releases.

**Review by safern — APPROVED** [review]

**Pull request merged**